packages/cdktf/lib/app.ts

```diff
@@ -63,37 +73,66 @@ export class App extends Construct {
 
     this.loadContext(options.context);
 
-    const node = Node.of(this);
+    const node = this.node;
     if (options.stackTraces === false) {
-      node.setContext(ConstructMetadata.DISABLE_STACK_TRACE_IN_METADATA, true);
+      node.setContext(DISABLE_STACK_TRACE_IN_METADATA, true);
     }
 
     node.setContext("cdktfVersion", version);
+
+    if (!fs.existsSync(this.outdir)) {
+      fs.mkdirSync(this.outdir);
+    }
+    this.manifest = new Manifest(version, this.outdir);
   }
 
   /**
    * Synthesizes all resources to the output directory
    */
   public synth(): void {
-    if (!fs.existsSync(this.outdir)) {
-      fs.mkdirSync(this.outdir);
-    }
-
-    const manifest = new Manifest(version, this.outdir);
-
-    Node.of(this).synthesize({
+    const session: ISynthesisSession = {
       outdir: this.outdir,
       skipValidation: this.skipValidation,
-      sessionContext: {
-        manifest,
-      },
+      manifest: this.manifest,
+    };
+
+    invokeAspects(this);
```

Comment:
Why on app level?

Reply:
To also support Annotations on the App (which might not make sense / might not be useful at all)

Reply:
Hm, probably not - since the App is mostly a container for stacks.